File: All Algorithms/Sorting Algorithm in Python/Recursive_Algorithms/insertion_sort.py
def insertionSortRecursive(array,n):
	if n == 1:
		return
	insertionSortRecursive(array,n-1)
	last = array[n-1]
	j = n-2
	while (j>=0 and array[j]>last):
		array[j+1] = array[j]
		j = j-1
	array[j+1]=last
	
array = [12,11,13,5,6]
insertionSortRecursive(arr, len(array))
print(array)

File: All Algorithms/Sorting Algorithm/INSERTION SORT.cpp
void duplicateZeros(vector<int>& arr) 
    {
        int n = arr.size();
        for (int i=0;i<n;i++)
        {
            if(arr[i]==0)
            {
                for(int j=n-2;j>=i;j--)
                {
                    arr[j+1]=arr[j];
                }
                i++;
            }
        }
        
    }